src/chocl8.zig

```diff
@@ -0,0 +1,303 @@
+            0x9 => {
+                std.log.info("{}: jump pc+2 ({}) if r[{}] ({}) != r[{}] ({})", .{ self.pc - 2, self.pc + 2, inst.x, self.reg[inst.x], inst.y, self.reg[inst.y] });
+                if (self.reg[inst.x] != self.reg[inst.y]) {
+                    self.pc += 2;
+                }
+            },
+            0xA => {
+                std.log.info("{}: index={}", .{ self.pc - 2, inst.nnn });
+                self.index = inst.nnn;
+            },
+            0xB => {
+                std.log.info("{}: jump r[0] ({}) + {}", .{ self.pc - 2, self.reg[0], inst.nnn });
+                self.pc = self.reg[0] + inst.nnn;
+            },
+            0xC => {
+                const rand = prng.random().int(u8);
+                std.log.info("{}: r[{}] = rand ({}) & {}", .{ self.pc - 2, inst.x, rand, inst.nn });
+                self.reg[inst.x] = inst.nn & rand;
+            },
+            0xD => {
+                // draw sprite
+                std.log.info("{}: drawing mem[{}] with h={}px @(r[{}]{}, r[{}]{})", .{ self.pc - 2, self.index, inst.n, inst.x, self.reg[inst.x], inst.y, self.reg[inst.y] });
+                const height: u8 = inst.n;
+                const x: u8 = @truncate(u6, self.reg[@truncate(u4, inst.x)] % 64);
+                const y: u8 = self.reg[@truncate(u4, inst.y)] % 32;
+                var i: usize = 0;
+                while ((i < height) and (y + i < 32)) : (i += 1) {
+                    self.fbuf[y + i] ^= @intCast(u64, self.mem[self.index + i]) << @truncate(u6, 8 * 7 - x);
+                }
+            },
+            0xE => switch (inst.nn) {
+                0x9E => {
+                    std.log.info("{}: jump pc+2 ({}) if keys[r[{}] ({})]", .{ self.pc - 2, self.pc + 2, inst.x, self.reg[inst.x] });
+                    if (0 != (keys & (@intCast(u16, 1) << @truncate(u4, self.reg[inst.x])))) {
+                        self.pc += 2;
+                    }
+                },
+                0xA1 => {
+                    std.log.info("{}: jump pc+2 ({}) if !keys[r[{}] ({})]", .{ self.pc - 2, self.pc + 2, inst.x, self.reg[inst.x] });
+                    if (0 == (keys & (@intCast(u16, 1) << @truncate(u4, self.reg[inst.x])))) {
+                        self.pc += 2;
+                    }
+                },
+                else => {
+                    @panic("0xE*NN variant uknown");
+                },
+            },
+            0xF => switch (inst.nn) {
+                0x07 => {
+                    std.log.info("{}: reg[{}] = delay ({})", .{ self.pc - 2, inst.x, self.delay });
+                    self.reg[inst.x] = self.delay;
+                },
+                0x0A => {
+                    std.log.info("{}: wait k; reg[{}] = k", .{ self.pc - 2, inst.x });
+                    if (keys != 0) {
+                        var i: u5 = 0;
+                        while (i < 16) : (i += 1) {
+                            if ((keys & (@intCast(u16, 1) << @truncate(u4, i))) != 0) {
```

Comment:
Also note here, `@as(u16, 1)` would be less verbose, and have the same effect. Integer literals are `comptime_int`, which can always be coerced to any type that can fit their value.